lib/exnumerable.ex

```diff
@@ -7,10 +7,18 @@ defmodule Exnumerator do
       def values, do: unquote(values)
       def sample, do: unquote(values) |> Enum.random
 
-      for value <- values do
-        def cast(unquote(value)), do: {:ok, unquote(value)}
-        def load(unquote(value)), do: {:ok, unquote(value)}
-        def dump(unquote(value)), do: {:ok, unquote(value)}
+      if Enum.all?(values, &(is_atom(&1))) do
```

Comment:
What happens when types are mixed?

Reply:
will get :error, nothing change for this, the same behavior